mllib/src/test/scala/org/apache/spark/ml/feature/VectorIndexerSuite.scala

```diff
@@ -121,10 +119,17 @@ class VectorIndexerSuite extends SparkFunSuite with MLlibTestSparkContext
 
     model.transform(densePoints1) // should work
     model.transform(sparsePoints1) // should work
-    intercept[SparkException] {
+    // If the data is local Dataset, it throws AssertionError directly.
+    intercept[AssertionError] {
```

Comment:
Yes, this is indeed a problem, let's try to find out all similar cases and resolved them in a separate PR. Thanks.